rest/index.py

```diff
@@ -468,6 +481,433 @@ def createIncident():
+        * `"modified"` is the date & time at which the chain was last modified.
+
+    Both the `"created"` and `"modified"` fields represent UTC timestamps and
+    are formatted like `YYYY/mm/dd HH:MM:SS`.
+    """
+
+    response.content_type = "application/json"
+
+    initConfig()
+
+    mongo = MongoClient(options.mongohost, options.mongoport)
+    dupchains = mongo.meteor[DUP_CHAIN_DB]
+
+    def _error(msg):
+        return json.dumps({
+            "error": msg,
+            "identifiers": [],
+            "created": toUTC(datetime.utcnow()).isoformat(),
+            "modified": toUTC(datetime.utcnow()).isoformat(),
+        })
+
+    query = {"alert": request.query.alert, "user": request.query.user}
+
+    if query["alert"] is None or query["user"] is None:
+        response.status = StatusCode.BAD_REQUEST
+        response.body = _error("Request missing `alert` or `user` field")
+        return response
+
+    chain = dupchains.find_one(query)
+
+    if chain is None:
+        # This is not an error, but we do want to write an empty response.
+        response.status = StatusCode.OK
+        response.body = _error("Did not find requested duplicate chian")
+        return response
+
+    response.status = StatusCode.OK
+    response.body = json.dumps(
+        {
+            "error": None,
+            "identifiers": chain["identifiers"],
+            "created": toUTC(chain["created"]).isoformat(),
+            "modified": toUTC(chain["modified"]).isoformat(),
+        }
+    )
+
+    return response
+
+
+@post("/alerttriagechain")
+@post("/alerttriagechain/")
+def create_duplicate_chain():
+    """Create a 'Duplicate Chain', linking information about alerts being
+    handled by the Triage Bot so that a user's response to a message about
+    one alert can be replicated against duplicate alerts without sending
+    multiple messages.
+
+    Requests are expected to take the following (JSON) form:
+
+    ```
+    {
+        "alert": str,
+        "user": str,
+        "identifiers": List[str]
+    }
+    ```
+
+    Where:
+        * `"alert"` is the "label" for the alert, signifying which of the
+        supported alerts is being triaged.
+        * `"user"` is the email address of the user contacted.
+        * `"identifier"` is a list of ElasticSearch IDs of alerts of the
+        same kind triggered by the same user.
+
+    This function writes back a response containing the following JSON.
+
+    ```
+    {
+        "error": Optional[str]
+    }
+    ```
+
+    If an error occurs, a duplicate chain will not be created and an error
+    string will be returned.  Otherwise, the `error` field will be `null.`
+    """
+
+    response.content_type = "application/json"
+
+    initConfig()
+
+    mongo = MongoClient(options.mongohost, options.mongoport)
+    dupchains = mongo.meteor[DUP_CHAIN_DB]
+
+    try:
+        req = json.loads(request.body.read())
+        request.body.close()
```

Comment:
You could use [bottle's native support for POSTed JSON](https://bottlepy.org/docs/dev/api.html#bottle.BaseRequest.json)